test/lib/mayaUsd/fileio/testSchemaApiAdaptor.py

```diff
@@ -45,9 +253,221 @@ def tearDownClass(cls):
     def setUp(self):
         pass
 
-    def testSimple(self):
-        mayaUsdLib.SchemaApiAdaptor.Register(schemaApiAdaptorTest, "mesh", "schemaApiName")
-        pass
+    def testMinimalAdaptation(self):
+        """Test that we can adapt ShadowAPI to an existing light shape. This exercises the most
+        basic callbacks exposed by a SchemaAPI adaptor:"""
+
+        mayaUsdLib.SchemaApiAdaptor.Register(shadowApiAdaptorShape, "light", "ShadowAPI")
+
+        lightShape1 = cmds.pointLight()
+        cmds.setAttr(lightShape1 + ".shadowColor", 0.5, 0.25, 0)
+        lightShape2 = cmds.pointLight()
+        cubeXform, cubeShape = cmds.polyCube()
+
+        # Wrong type: not adapted
+        adaptor = mayaUsdLib.Adaptor(cubeShape)
+        self.assertEqual(adaptor.GetAppliedSchemas(), [])
+
+        # Wrong name: not adapted
+        adaptor = mayaUsdLib.Adaptor(lightShape2)
+        self.assertEqual(adaptor.GetAppliedSchemas(), [])
+
+        # Adapted:
+        adaptor = mayaUsdLib.Adaptor(lightShape1)
+        self.assertEqual(adaptor.GetAppliedSchemas(), ["ShadowAPI"])
+
+        schema = adaptor.GetSchemaByName("ShadowAPI")
+        self.assertTrue(schema)
+
+        self.assertEqual(schema.GetAuthoredAttributeNames(),
+                         ["inputs:shadow:color", "inputs:shadow:enable"])
+        colorAttr = schema.GetAttribute("inputs:shadow:color")
+        self.assertTrue(colorAttr)
+        linearizedValue = (0.21763764, 0.047366142, 0)
+        colorAttrValue = colorAttr.Get()
+        self.assertAlmostEqual(linearizedValue[0], colorAttrValue[0])
+        self.assertAlmostEqual(linearizedValue[1], colorAttrValue[1])
+        self.assertEqual(linearizedValue[2], colorAttrValue[2])
+
+        colorAttr.Set((1,0,1))
+        self.assertEqual(cmds.getAttr(lightShape1 + ".shadowColor"), [(1.0, 0.0, 1.0)])
+
+    @unittest.skipUnless(HAS_BULLET, 'Requires the bullet plugin.')  
+    def testComplexAdaptation(self):
+        """Test that we can adapt a bullet simulation"""
+
+        mayaUsdLib.SchemaApiAdaptor.Register(TestBulletMassShemaAdaptor, "shape", "PhysicsMassAPI")
+        mayaUsdLib.SchemaApiAdaptor.Register(TestBulletRigidBodyShemaAdaptor, "shape", "PhysicsRigidBodyAPI")
+
+        # Build a scene (and exercise the adaptor in a freeform context)
+        cmds.file(f=True, new=True)
```

Comment:
Same code as found in `test\lib\usd\translators\testUsdExportSchemaApi.py` meaning the Python code can do what the C++ code does.